server/src/main/scala/de/frosner/broccoli/models/ParameterValue.scala

```diff
@@ -0,0 +1,82 @@
+package de.frosner.broccoli.models
+
+import de.frosner.broccoli.services.{ParameterValueParsingException, TemplateParameterNotFoundException}
+import org.apache.commons.lang3.StringEscapeUtils
+import play.api.libs.json._
+
+import scala.util.{Failure, Success, Try}
+
+
+object ParameterValue {
+
+    implicit val parameterValueReads: Reads[ParameterValue] = new Reads[ParameterValue] {
+        override def reads(
+            json: JsValue): JsResult[ParameterValue] = ???
+    }
+
+    def constructParameterValueFromJson(parameterName: String,
+                              template: Template,
+                              jsValue: JsValue): Try[ParameterValue] = {
+        Try{
+            template.parameterInfos.get(parameterName) match {
+                case Some(parameterInfo) =>
+                    constructParameterValueFromJson(parameterInfo.`type`.getOrElse(ParameterType.Raw), jsValue) match {
+                        case Some(param) => param
+                        case None =>
+                            throw ParameterValueParsingException(template.id, parameterName)
+                    }
+                case None =>
+                    throw TemplateParameterNotFoundException(template.id, parameterName)
+            }
+        }
+    }
+
+    def constructParameterValueFromJson(parameterType: ParameterType, jsValue: JsValue): Option[ParameterValue] = {
+        Try {
+            parameterType match {
+                case ParameterType.Raw =>
+                    RawParameterValue(jsValue.as[String])
+                case ParameterType.String =>
+                    StringParameterValue(jsValue.as[String])
+                case ParameterType.Integer =>
+                    IntParameterValue(jsValue.as[Int])
+                case ParameterType.Decimal =>
+                    DecimalParameterValue(jsValue.as[BigDecimal])
+            }
+        } match {
+            case Success(s) => Some(s)
+            case Failure(_) => None
+        }
+    }
+}
+trait ParameterValue {
+    def isEmpty: Boolean
```

Comment:
What is the `isEmpty` method used for?

Reply:
Not sure. Used to remove params with zero length strings from the paramvalues map before. Maybe won't be required anymore after frontend changes.